### PR TITLE
Minimal version to upload files

### PR DESCRIPTION
**Pull request by leplatrem**
I did a small script that could be useful to populate and update the server based on the content of a folder for example.

@Natim @tarekziade thoughts ?

### CODE REVIEW

**Comment by Natim** [comment]
Nice :+1:

**Comment by leplatrem** [comment]
I wrote the download counterpart.

@Natim @almet r?

File: scripts/download.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+                continue
+
+        # Download remote attachment by chunk.
+        resp = session.get(attachment['location'], stream=True)
+        resp.raise_for_status()
+        tmp_file = destination + '.tmp'
+        with open(tmp_file, 'wb') as f:
+            for chunk in resp.iter_content(chunk_size=chunk_size):
+                if chunk:
+                    f.write(chunk)
```
Comment (Natim) [inline scripts/download.py]: You could probably calculate the hash here too

File: scripts/download.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+        # Decompress the file if necessary.
+        if is_gzip:
+            with open(destination, 'wb') as f:
+                data = open(tmp_file, 'rb').read()
+                f.write(gzip.decompress(data))
```
Comment (Natim) [inline scripts/download.py]: Unless if it compressed.

File: scripts/download.py [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+                        help='Collection URL', required=True)
+    parser.add_argument('--auth', dest='auth', action='store',
+                        help='Credentials')
+    parser.add_argument('--folder', dest='folder', action='store',
+                        default='.', help='Destination folder')
+    args = parser.parse_args()
+
+    session = requests.Session()
+    if args.auth:
+        session.auth = tuple(args.auth.split(':'))
```
Comment (Natim) [inline scripts/download.py]: If there are no ":" in the auth we should use [getpass](https://docs.python.org/2/library/getpass.html) to ask the password.

See also: https://github.com/mozilla/PyFxA/blob/master/fxa/__main__.py#L142-L148

File: scripts/download.py [hunk 4]
```diff
@@ -0,0 +1,110 @@
...
+
+    url = args.url
+    if url.endswith('/'):
+        url = url[:-1]
```
Comment (Natim) [inline scripts/download.py]: what about using rstrip()?

File: scripts/download.py [hunk 5]
```diff
@@ -0,0 +1,110 @@
...
+
+def sha256(content):
+    m = hashlib.sha256()
+    m.update(content)
+    return m.hexdigest()
+
+
+def fetch_records(session, url, since=None):
+    if since:
+        url = url + "?_since=%s" % since
```
Comment (Natim) [inline scripts/download.py]: You should add a ``_sort=-last_modified`` parameter here according to the way you store the timestamp.

File: README.rst [hunk 6]
```diff
@@ -206,6 +206,30 @@ Using JavaScript
...
+    $ python3 scripts/upload.py --url $COLLECTION_URL --auth "user:pass" --gzip README.rst pictures/*
+
+See ``python scripts/upload.py --help`` for more details about options.
+
+Download files
+--------------
+
+::
+
+    $ COLLECTION_URL=https://kinto.dev.mozaws.net/v1/buckets/fennec-ota/collections/locale/records
```
Comment (Natim) [inline README.rst]: What about adding ``--bucket`` ``--collection`` and ``--host`` instead ?

File: scripts/download.py [hunk 7]
```diff
@@ -0,0 +1,110 @@
...
+        url = url[:-1]
+    if not url.endswith('records'):
+        url += '/records'
+
+    # See if timestamp was saved from last run.
+    last_sync = None
+    timestamp = os.path.join(args.folder, '.last_sync')
+    if os.path.exists(args.folder):
+        if os.path.exists(timestamp):
+            last_sync = open(timestamp, 'r').read()
```
Comment (Natim) [inline scripts/download.py]: Don't want to validate that the value is accurate to be a ``_since`` parameter?

File: scripts/upload.py [hunk 8]
```diff
@@ -0,0 +1,121 @@
...
+
+    # XXX: add option to delete records when files are missing locally
+    for id, record in records_by_id.items():
+        print("Ignore remote file '%s'." % record['attachment']['filename'])
+
+    return to_upload
+
+
+def upload_files(session, url, files, compress):
+    permissions = {}  # XXX not set yet
```
Comment (Natim) [inline scripts/upload.py]: We probably want permissions on the collection for this usecase.

File: scripts/upload.py [hunk 9]
```diff
@@ -0,0 +1,121 @@
...
+        response = session.post(attachment_uri, data=payload, files=multipart)
+        response.raise_for_status()
+        pprint.pprint(response.json())
+
+
+def main():
+    parser = argparse.ArgumentParser(description='Upload files to Kinto')
+    parser.add_argument('--url', dest='url', action='store',
+                        help='Collection URL', required=True)
+    parser.add_argument('--auth', dest='auth', action='store',
```
Comment (Natim) [inline scripts/upload.py]: Same proposal for the auth.

**Pull request closed without merging**